### PR TITLE
[TT-4839] Add x-tyk-api-gateway-schema in OAS validator

### PR DESCRIPTION
**Pull request by jeffy-mathew**
[changelog]
Added: x-tyk-api-gateway schema in OAS validator

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3996/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2109328830)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3996/merge
Commit: 030b995100f79f0f576a4ba0a3d44c10d5bcccb2 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2110521359)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3996/merge
Commit: 87edc8c0f7d3e2ee20b4cc7069ec0188e1ac6855 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2112960260)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3996/merge
Commit: 51501071929b4b4a210f04c9afc3ba3f97e942ce 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2113852635)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3996/merge
Commit: d16bd1a0ef2680ffb668f6c86a01febfebcc0736 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2117121387)

**Review by titpetric — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3996/merge
Commit: b7c5220fd7efadfdddda3bdbf91c7fde1943afd5 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2147335969)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3996/merge
Commit: 526b93975cb0bd5c9f20089e32030da952634211 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2147464554)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3996/merge
Commit: 5a3e8933bf068383243a7d1e1a7f0a73a280b6ab 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2152841091)

File: apidef/oas/schema/x-tyk-api-gateway.json [hunk 1]
```diff
@@ -0,0 +1,915 @@
...
+          "type": "string"
+        }
+      },
+      "required": [
+        "enabled"
+      ]
+    },
+    "X-Tyk-Certificate": {
+      "type": "object",
+      "properties": {
```
Comment (furkansenharputlu) [inline apidef/oas/schema/x-tyk-api-gateway.json:78]: both domain and certificate are required here

File: apidef/oas/schema/x-tyk-api-gateway.json [hunk 2]
```diff
@@ -0,0 +1,915 @@
...
+        "goPlugin": {
+          "$ref": "#/definitions/X-Tyk-GoPlugin"
+        },
+        "customPlugin": {
+          "$ref": "#/definitions/X-Tyk-CustomPlugin"
+        },
+        "securitySchemes": {
+          "type": "object",
+          "^\\/": {
+            "anyOf": [
```
Comment (furkansenharputlu) [inline apidef/oas/schema/x-tyk-api-gateway.json:530]: shouldn't it be `oneOf`? Native one uses that.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3996/merge
Commit: ed4166fba9dc833317fbaa8b7085fc8b358d7fa9 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2165529332)

File: apidef/oas/schema/x-tyk-api-gateway.json (hunk 2 above)
Comment (jeffy-mathew) [reply apidef/oas/schema/x-tyk-api-gateway.json]: updated

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3996/merge
Commit: fed68513528c01209d35c10cb7c50e7f78abb1db 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2182347869)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3996/merge
Commit: f3eb7b89d9cc6f6cb83e899402bcb4337d9ace9f 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2182357550)

**Review by titpetric — APPROVED** [review]

**Review by furkansenharputlu — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 5b49f0441db680c23a7b85e0d7565ae2a017bc45 [TT-4839] Add x-tyk-api-gateway-schema in OAS validator (#3996)

* Add x-tyk-api-gateway-schema in OAS validator

[changelog]
Added: x-tyk-api-gateway schema in OAS validator
Triggered by: push (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2183851408)

**Pull request merged**